python_modules/dagster/dagster/_core/definitions/partitioned_schedule.py

```diff
@@ -144,15 +149,23 @@ def do_stuff_partitioned():
             check.failed("The provided job is not partitioned")
 
         partitions_def = _check_valid_schedule_partitions_def(partitions_def)
-        time_partitions_def = check.not_none(get_time_partitions_def(partitions_def))
+        if isinstance(partitions_def, StaticPartitionsDefinition):
+            check.not_none(
+                cron_schedule,
+                "Creating a schedule from a static partitions definition requires a cron schedule",
+            )
+        else:
+            time_partitions_def = check.not_none(get_time_partitions_def(partitions_def))
+            cron_schedule = cron_schedule or time_partitions_def.get_cron_schedule(
+                minute_of_hour, hour_of_day, day_of_week, day_of_month
+            )
+            execution_timezone = execution_timezone or time_partitions_def.timezone
```

Comment:
could we also make an assertion here that `execution_timezone` is none and just rely on `time_partitions_def.timezone`?

If the job is time partitioned, I find it hard to imagine use cases for providing a different time zone